internal/builders/maven/plugins/verification-plugin/src/main/java/io/github/adamkorcz/SlsaVerificationMojo.java

```diff
@@ -0,0 +1,189 @@
+                    mavenSession,
+                    pluginManager
+                )
+            );
+        } catch(MojoExecutionException e) {
+            getLog().info("Skipping slsa verification: Fail to retrieve slsa-verifier.");
+            return;
+        }
+
+        // Verify the slsa of each dependency
+        Set<Artifact> dependencyArtifacts = project.getDependencyArtifacts();
+        for (Artifact artifact : dependencyArtifacts ) {
+            // Retrieve the dependency jar and its slsa file
+            String artifactStr = artifact.getGroupId() + ":" + artifact.getArtifactId() + ":" + artifact.getVersion();
+            try {
+                // Retrieve the slsa file of the artifact
+                executeMojo(
+                    plugin(
+                        groupId("com.googlecode.maven-download-plugin"),
+                        artifactId("download-maven-plugin"),
+                        version("1.7.0")
+                    ),
+                    goal("artifact"),
+                    configuration(
+                        element(name("outputDirectory"), "${project.build.directory}/slsa"),
+                        element(name("groupId"), artifact.getGroupId()),
+                        element(name("artifactId"), artifact.getArtifactId()),
+                        element(name("version"), artifact.getVersion()),
+                        element(name("type"), "intoto.build.slsa"),
+                        element(name("classifier"), "jar")
+                    ),
+                    executionEnvironment(
+                        project,
+                        mavenSession,
+                        pluginManager
+                    )
+                );
+
+                // Retrieve the dependency jar if slsa file does exists for this artifact
+                executeMojo(
+                    plugin(
+                        groupId("org.apache.maven.plugins"),
+                        artifactId("maven-dependency-plugin"),
+                        version("3.6.0")
+                    ),
+                    goal("copy"),
+                    configuration(
+                        element(name("outputDirectory"), "${project.build.directory}/slsa"),
+                        element(name("artifact"), artifactStr)
+                    ),
+                    executionEnvironment(
+                        project,
+                        mavenSession,
+                        pluginManager
+                    )
+                );
+            } catch(MojoExecutionException e) {
+                getLog().info("Skipping slsa verification for " + artifactStr + ": No slsa file found.");
+                continue;
+            }
+
+            // Verify slsa file
+            try {
+                // Run slsa verification on the artifact and print the result
+                // It will never fail the build process
+                String arguments = "verify-artifact --provenance-path ";
```

Comment:
Can we be careful about script injection? Is there an exec (non-shell) command we can use instead?

Reply:
Looks like the arguments are split by the exec-maven-plugin using `CommandLineUtils.tranlateCommandLine` which is implemented here: https://github.com/sonatype/plexus-utils/blob/5ba6cfcca911200b5b9d2b313bb939e6d7cbbac6/src/main/java/org/codehaus/plexus/util/cli/CommandLineUtils.java#L302-L383

It doesn't look obviously vulnerable to me but it does make me wary that the command is taken as a String rather than a String array and then split this way.

Reply:
Would prefer to catch up on this in a separate PR to be sure that we properly guard against command injections here. I have added an inline comment with a TODO.

Reply:
sg, let's differ to the follow-up PR. Please create a tracking issue and update the comment to `TODO(#number)`